app/restaurant/controller.py

```diff
@@ -6,49 +6,221 @@
+            },
+            'data' : {
+                'restaurants' : restaurants
+            }
+        })
+
+    else: # POST request
+        try:
+            request.json["restaurant_number"] = int(request.json["restaurant_number"])
+            print(request.json['restaurant_number'])
+        except:
+            return jsonify({
+                'status' : 'error',
+                'errors' : [
+                    'Restaurant number must be a positive integer.'
+                ],
+                'message' : 'There was a problem making the request.'
             }), 400
 
-        restaurant = Restaurant(user, request.json['name'], request.json['restaurant_number'], request.json['address'])
+        request.json['owner_id'] = g.user['id']
+
+        form = RestaurantValidator(data=request.json)
+
+        if form.validate():
+
+            # todo get user from the token
+            user = User.query.filter(User.id==g.user['id']).all()
+
+            restaurant = Restaurant(user, request.json['name'], request.json['restaurant_number'], request.json['address'])
+            user.role = USER.OWNER
 
-        db.session.add(restaurant)
-        db.session.commit()
+            db.session.add(restaurant)
+            db.session.commit()
+
+            return jsonify({
+                'status' : 'success',
+                'message' : 'Restaurant successfully added',
+                'data' : {
+                    'restaurant' : restaurant.to_dict()
+                }
+            }), 201
 
         return jsonify({
-            'message' : 'Restaurant successfully added',
+            'message' : 'There is missing data',
+            'errors' : form.errors
+        }), 400
+
+
+@restaurantMod.route('/<int:restaurantId>', methods=['GET', 'PUT', 'DELETE'])
+@requires_login
+def restaurant_view(restaurantId):
+    # Happens for every request on this route
+    if restaurantId < 1:
+        return jsonify({
+        'message' : 'There was an error',
+        'errors' : {
+        'restaurantId' : 'Must be greater than 0'
+        }
+        }), 404
+
+    restaurants = Restaurant.query.filter(Restaurant.id == restaurantId).all()
+
+    if len(restaurants) is not 1:
+        return jsonify({
+        'message' : 'Restaurant cannot be found',
+        'errors' : {
+        'restaurant' : 'No restaurant with id of %d' % restaurantId
+        }
+        }), 404
+
+    if request.method == 'GET': # single view
+        restaurant = restaurants[0].to_dict()
+        restaurant['owner'] = restaurants[0].owner.to_dict()
+
+        return jsonify({
+            'status' : 'success',
+            'message' : 'restaurant found',
             'data' : {
-                'restaurant' : restaurant.to_dict()
+                'restaurant' : restaurant
             }
-        }), 201
+        })
+    if request.method == 'DELETE': # single delete
+        if restaurants[0].owner.id != g.user['id'] and g.user['role'] != 'admin':
```

Comment:
`owner_id = restaurants[0].owner.id
user_id = g.user['id']
user_role = g.user['role']

owner = user_id == owner_id
admin = user_rold == 'admin'
if not owner and not admin:`

..or..
`owner = g.user['id'] == restaurants[0].owner.id
admin = g.user['role'] == 'admin'

if not owner and not admin:
`
Both changes are readability changes. 
The former is preferred over the later, however, both are improvements readability wise. 
Ad added perk is smaller bottleneck on the the `if` evaluation.